com.github.falkTX.Carla.json

```diff
@@ -0,0 +1,134 @@
+{
+  "app-id": "com.github.falkTX.Carla",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15-21.08",
+  "sdk": "org.kde.Sdk",
+  "command": "carla",
+  "rename-desktop-file": "carla.desktop",
+  "rename-icon": "carla",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    /* MIDI */
+    "--device=all",
+    /* For pipewire */
+    "--filesystem=xdg-run/pipewire-0",
+    "--system-talk-name=org.freedesktop.RealtimeKit1",
+    /* Allow loading, saving files from the home directory (portals don’t work yet) */
+    "--filesystem=home",
+    "--env=TMPDIR=/var/tmp",
+    /* On SuSE it seems to be necessary for sound to work */
+    "--env=ALSA_CONFIG_PATH=",
+    "--env=LV2_PATH=$HOME/.lv2:/app/extensions/Plugins/lv2",
+    "--env=DSSI_PATH=/app/extensions/Plugins/dssi",
+    "--env=VST_PATH=/app/extensions/Plugins/lxvst",
+    "--env=VST3_PATH=$HOME/.vst3:/app/extensions/Plugins/vst3",
+    "--env=LADSPA_PATH=/app/extensions/Plugins/ladspa"
+  ],
+  "add-extensions": {
+    "org.freedesktop.LinuxAudio.Plugins": {
+      "directory": "extensions/Plugins",
+      "version": "21.08",
+      "add-ld-path": "lib",
+      "merge-dirs": "ladspa;lv2;lxvst;vst3",
+      "subdirectories": true,
+      "no-autodownload": true
+    }
+  },
+  "cleanup": [
+    "/include",
+    "/man",
+    "/share/man",
+    "/share/info",
+    "/lib/pkgconfig",
+    "*.la"
+  ],
+  "build-options": {
+    "env": {
+      "PIP_DISABLE_PIP_VERSION_CHECK": "1"
+    }
+  },
+  "modules": [
+    "shared-modules/linux-audio/jack2.json",
+    "shared-modules/linux-audio/fluidsynth2.json",
+    {
+      "name": "pyqt5",
+      "config-opts": [
+        "--disable-static",
+        "--enable-x11"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://files.pythonhosted.org/packages/4d/81/b9a66a28fb9a7bbeb60e266f06ebc4703e7e42b99e3609bf1b58ddd232b9/PyQt5-5.14.2.tar.gz",
+          "sha256": "bd230c6fd699eabf1ceb51e13a8b79b74c00a80272c622427b80141a22269eb0"
+        },
+        {
+          "type": "script",
+          "commands": [
+            "python3 configure.py --assume-shared --confirm-license --no-designer-plugin --no-qml-plugin --sysroot=/app --qsci-api --qsci-api-destdir=/app/qsci --sipdir=/app/share/sip --sip=/app/bin/sip --sip-incdir=/app/include QMAKE_CFLAGS_RELEASE='-I/usr/include/python3.9/' QMAKE_CXXFLAGS_RELEASE='-I/usr/include/python3.9/'"
```

Comment:
Please look at how I [packaged pyqt](https://github.com/flathub/org.qutebrowser.qutebrowser/blob/master/pyqt5/pyqt5.json), and the [configure script](https://github.com/flathub/org.qutebrowser.qutebrowser/blob/master/pyqt5/sip/sip-build-configure.in). This will help you avoid manually changing the Python version string when switching to a new runtime version.

Reply:
maybe we should make a shared-module for this.

Reply:
I started working on this, but I'm not sure what is the best way to let maintainer choose different configure options (enabled PyQt bindings) while not breaking automated updates/f-e-d-c of Flathub's shared-modules or the app packaging.

Reply:
We shouldn't need to choose different configure option. You can't when using distro package.

Reply:
Looking at all the files needed for pyqt5 in your flatpak, this looks way too complicated for building one simple dependency. Also I'm still a beginner to all of this, so not sure what exactly is really needed for Carla (to copy the pyqt5 build files from your project may be too much right?).